boot/boot_serial/src/boot_serial.c

```diff
@@ -714,5 +721,35 @@ boot_serial_start(const struct boot_uart_funcs *f)
             boot_serial_input(&dec_buf[2], dec_off - 2);
         }
         off = 0;
+check_timeout:
+        /* Subtract elapsed time */
+#ifdef MCUBOOT_SERIAL_WAIT_FOR_DFU
+        timeout_in_ms -= (k_uptime_get_32() - start);
+#endif
```

Comment:
I guess it is OK, although usage of Zephyr specific k_ call here, in common code, drops system agnostic solution on the next dev that enables `MCUBOOT_SERIAL_WAIT_FOR_DFU` for other system.